ratik/src/main/java/org/opencadc/ratik/InventoryValidator.java

```diff
@@ -0,0 +1,146 @@
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  : 5 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.ratik;
+
+import ca.nrc.cadc.db.DBUtil;
+import ca.nrc.cadc.net.ResourceNotFoundException;
+import ca.nrc.cadc.reg.Capabilities;
+import ca.nrc.cadc.reg.Capability;
+import ca.nrc.cadc.reg.Standards;
+import ca.nrc.cadc.reg.client.RegistryClient;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+import javax.sql.DataSource;
+import org.apache.log4j.Logger;
+import org.opencadc.inventory.InventoryUtil;
+import org.opencadc.inventory.db.ArtifactDAO;
+import org.opencadc.inventory.db.version.InitDatabase;
+
+/**
+ * Validate local inventory.
+ */
+public class InventoryValidator implements Runnable {
+    private static final Logger log = Logger.getLogger(InventoryValidator.class);
+
+    private final ArtifactDAO artifactDAO;
+    private final URI resourceID;
+    private final String uriBuckets;
+
+    /**
+     * Constructor.
+     *
+     * @param daoConfig          config map to pass to cadc-inventory-db DAO classes
+     * @param resourceID         identifier for the remote query service
+     * @param uriBuckets         uri buckets
+     */
+    public InventoryValidator(Map<String, Object> daoConfig, URI resourceID, String uriBuckets) {
+        InventoryUtil.assertNotNull(InventoryValidator.class, "daoConfig", daoConfig);
+        InventoryUtil.assertNotNull(InventoryValidator.class, "resourceID", resourceID);
+        InventoryUtil.assertNotNull(InventoryValidator.class, "selector", uriBuckets);
+
+        this.artifactDAO = new ArtifactDAO(false);
+        this.artifactDAO.setConfig(daoConfig);
+        this.resourceID = resourceID;
+        this.uriBuckets = uriBuckets;
+
```

Comment:
Something missing here: ratik needs to mimic fenwick in selecting AllArtifacts or an ad-hoc filter policy. That was supported by a class named IncludeArtifacts. That means moving some code so it is re-usable here because it really does have to be the same and operators really have to put the same artifact-filter.sql file into fenwick and ratik or they will fight over what is to be included. This is needed for every discrepancy that mentions "filter policy".
* move code from fenwick to cadc-inventory-util (org.opencadc.inventory.util)
* update README for new config item and additional config file

Reply:
Moved the ArtifactSelector classes into cadc-inventory-util and updated the README.